src/test/kotlin/com/dprint/services/editorservice/EditorServiceCacheTest.kt

```diff
@@ -0,0 +1,158 @@
+    }
+
+    test("canFormatCached when result exists returns result") {
+        // Given
+        val cache = EditorServiceCache(mockProject)
+        val path = "/test/file.ts"
+        cache.putCanFormat(path, true)
+
+        // When
+        val result = cache.canFormatCached(path)
+
+        // Then
+        result shouldBe true
+    }
+
+    test("canFormatCached when result does not exist returns null") {
+        // Given
+        val cache = EditorServiceCache(mockProject)
+        val path = "/test/file.ts"
+
+        // When
+        val result = cache.canFormatCached(path)
+
+        // Then
+        result shouldBe null
+    }
+
+    test("putCanFormat stores value in cache") {
+        // Given
+        val cache = EditorServiceCache(mockProject)
+        val path = "/test/file.ts"
+
+        // When
+        cache.putCanFormat(path, true)
+
+        // Then
+        cache.canFormatCached(path) shouldBe true
+    }
+
+    test("clearCanFormatCache removes all entries") {
+        // Given
+        val cache = EditorServiceCache(mockProject)
+        cache.putCanFormat("/test/file1.ts", true)
+        cache.putCanFormat("/test/file2.ts", false)
+
+        // When
+        cache.clearCanFormatCache()
+
+        // Then
+        cache.canFormatCached("/test/file1.ts") shouldBe null
+        cache.canFormatCached("/test/file2.ts") shouldBe null
+    }
+
+    test("createPrimeCanFormatTask when editor service null creates task with warning") {
+        // Given
+        val cache = EditorServiceCache(mockProject)
+        val path = "/test/file.ts"
+
+        // When
+        cache.createPrimeCanFormatTask(path, null, mockTaskQueue)
+
+        // Then
+        verify {
+            mockTaskQueue.createTaskWithTimeout(
+                match { it.taskType == TaskType.PrimeCanFormat && it.path == path },
+                any(),
+                any(),
+                5000L,
+            )
+        }
+    }
+
+    test("createPrimeCanFormatTask when editor service exists creates task and updates cache") {
+        // Given
+        val cache = EditorServiceCache(mockProject)
+        val path = "/test/file.ts"
+        coEvery { mockEditorService.canFormat(path) } returns true
+
+        // When
+        cache.createPrimeCanFormatTask(path, mockEditorService, mockTaskQueue)
+
+        // Then
+        verify {
+            mockTaskQueue.createTaskWithTimeout(
+                match { it.taskType == TaskType.PrimeCanFormat && it.path == path },
+                any(),
+                any(),
+                5000L,
+            )
+        }
+    }
+
+    test("cache uses LRU behavior") {
+        // Given
+        val cache = EditorServiceCache(mockProject)
+        for (i in 1..200) {
+            cache.putCanFormat("/test/file$i.ts", true)
+        }
+
+        // When - Access should work for recently added items
+        val result = cache.canFormatCached("/test/file200.ts")
+
+        // Then
+        result shouldBe true
+    }
```

Comment:
Not necessary unless we can actually argue about it dropping the oldest result. Probably best just removing the test since we dont want to test the internals of the LRU impl